src/main/java/com/talanlabs/sonar/plugins/gitlab/GlobalReport.java

```diff
@@ -109,27 +109,29 @@ private void appendExtraIssues(StringBuilder builder, boolean hasInlineIssues, b
 
         appendSeverities(builder);
     }
-
+    
     private void appendSeverities(StringBuilder builder) {
         int notReportedDisplayedIssueCount = 0;
-        int i = 0;
-
+        int reportedIssueCount = 0;
+        
         for (Severity severity : SEVERITIES) {
             List<IssueUrl> issueUrls = notReportedOnDiffMap.get(severity);
             if (issueUrls != null && !issueUrls.isEmpty()) {
                 for (IssueUrl issueUrl : issueUrls) {
                     PostJobIssue postJobIssue = issueUrl.postJobIssue;
-                    String msg = "1. " + markDownUtils.globalIssue(postJobIssue.severity(), postJobIssue.message(), postJobIssue.ruleKey().toString(), issueUrl.gitLabUrl, postJobIssue.componentKey());
-                    if (i < gitLabPluginConfiguration.maxGlobalIssues()) {
-                        builder.append(msg).append("\n");
+                    if (reportedIssueCount < gitLabPluginConfiguration.maxGlobalIssues()) {
+                        builder.append(reportedIssueCount)
```

Comment:
moved the markdownutil call into the if block to only create strings when necessary.